ros_buildfarm/workspace.py

```diff
@@ -41,6 +42,32 @@ def clean_workspace(workspace_root):
     if os.path.exists(test_results_dir):
         shutil.rmtree(test_results_dir)
 
+def call_abi_checker(workspace_root, rosdistro_name, env):
+    # TODO: pkgs detection, code based on create_devel_task_generator.py
+    condition_context = {}
+    condition_context['ROS_DISTRO'] = rosdistro_name
+
+    pkgs = {}
+    for ws_root in workspace_root:
+        source_space = os.path.join(ws_root, 'src')
+        ws_pkgs = find_packages(source_space)
+        for pkg in ws_pkgs.values():
+            pkg.evaluate_conditions(condition_context)
+        pkgs.update(ws_pkgs)
+    pkg_names = [pkg.name for pkg in pkgs.values()]
+
+    # TODO: workspace_root[0] to be compatible with a code in
+    # create_devel_task_generator for a future refactor. To fix
+    # it, implement the support for multiple local-dir in auto-abi tool
```

Comment:
Instead of a TODO comment how about asserting that `len(workspace_root) == 1`? That way if it is being used with multiple paths in the future it will actually fail instead of silently using only the first.

Reply:
good idea, 8641225